dashboard/app/jobs/aichat_request_chat_completion_job.rb

```diff
@@ -0,0 +1,80 @@
+class AichatRequestChatCompletionJob < ApplicationJob
+  queue_as :default
+
+  before_enqueue do |job|
+    request = job.arguments.first[:request]
+    request.update(execution_status: SharedConstants::AI_REQUEST_EXECUTION_STATUS[:QUEUED])
+  end
+
+  before_perform do |job|
+    request = job.arguments.first[:request]
+    request.update(execution_status: SharedConstants::AI_REQUEST_EXECUTION_STATUS[:RUNNING])
+  end
+
+  # Catch any exceptions that occur during the job and update the request status accordingly
+  rescue_from Exception do |exception|
+    if rack_env?(:development)
+      puts "AichatRequestChatCompletionJob Error: #{exception.full_message}"
+    end
+
+    request = arguments.first[:request]
+    request.update(response: exception.message, execution_status: SharedConstants::AI_REQUEST_EXECUTION_STATUS[:FAILURE])
+    Honeybadger.notify(
+      "AichatRequestChatCompletionJob failed with unexpected error: #{exception.message}",
+      context: {
+        request: request.to_json
+      }
+    )
+  end
+
+  def perform(request:, locale:)
```

Comment:
I did notice later in those docs there is the option to add a serializer for your class, which would allow you to keep passing the request object directly, in case that sounds preferable to you.

Reply:
Yeah, I originally had request ID, but then saw the docs below about how GlobalID allows passing ActiveRecord objects directly. Looks like as per the docs,  `this works with any class that mixes in GlobalID::Identification, which by default has been mixed into Active Record classes` so we may not even need the serializer?

That being said, I don't have a strong preference either way - if it's better code style to pass in the ID and lookup the object inside the job, that's fine with me too.

Reply:
oh, nice find! I missed that GlobalID included active record objects. if this is supported then I think it sounds great to pass the object directly.